### ISSUE TITLE
The Sign up button from the Connect with us section doesn't respond to mouse hovers

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-server#12700]
Follow up for #12613 

### Describe the problem and steps to reproduce it:
1. Open DevHub homepage in logged out view - https://addons.allizom.org/en-US/developers/
2. Scroll down to the Connect with us section (blue footer)
3. Hover over the Sign up button form the Extensions Developer Newsletter component

### What happened?
There is no response from the button on mouse hover (mouse pointer doesn't change)

### What did you expect to happen?
The pointer should change when hovering over the button to indicate an interaction

### Anything else we should know?
- I've added a comparison with Extension workshop to illustrate what the expected result should be

[DevHub]
![2019-10-23_09-20-27](https://user-images.githubusercontent.com/31961530/67364729-cca8b580-f578-11e9-89e8-5026fc11a8ab.gif)

[Extension workshop]
![2019-10-23_09-21-05](https://user-images.githubusercontent.com/31961530/67364739-d3372d00-f578-11e9-8177-b47afe85059b.gif)

**Comment by willdurand** [comment]
This "sign up" button is using the style of all the buttons in DevHub as per https://github.com/mozilla/addons-server/pull/12733#issuecomment-547010343. There is no `:hover`, `:focus` or `:active` support, so we should probably add that (+ `cursor: pointer`)

**Comment by BenTarman** [comment]
Hi I would like to get this assigned to me for my first contribution. Just to clarify, I plan on just make the styling similar to what we have on the Extension workshop page for the button.

**Comment by willdurand** [comment]
Thanks @BenTarman (cc @caitmuenster)

**Comment by ioanarusiczki** [comment]
@willdurand @BenTarman 
Looks good to me! Verified on AMO dev with FF70(win10). 

![button with more stype](https://user-images.githubusercontent.com/33448286/68766098-e6dd2d00-0626-11ea-8190-57725549d31d.gif)
